controller/tests/test_volunteer_controller.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+import controller.volunteer_controller as vc
+from models.volunteer import Volunteer
+from models.camp import Camp
+
+
+class TestVolunteerController(unittest.TestCase):
+
+    def test_create_volunteer(self):
+        volunteer_called = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                               phone='+447519953189', camp=Camp(name='UCL'))
+        self.assertEqual(type(volunteer_called), Volunteer)
+
+    def test_create_volunteer_invalid_username(self):
+        with self.assertRaises(vc.ControllerError):
+            vc.create_volunteer(username='yun', password='root', firstname='Yunsy', lastname='Yin',
+                                phone='+447519953189', camp=Camp(name='UCL'))
+
+    def test_create_volunteer_invalid_password(self):
+        with self.assertRaises(vc.ControllerError):
+            vc.create_volunteer(username='yunsy', password='roo', firstname='Yunsy', lastname='Yin',
+                                phone='+447519953189', camp=Camp(name='UCL'))
+
+    def test_create_volunteer_invalid_firstname(self):
+        with self.assertRaises(vc.ControllerError):
+            vc.create_volunteer(username='yunsy', password='roo', firstname='Y', lastname='Yin',
+                                phone='+447519953189', camp=Camp(name='UCL'))
+
+    def test_create_volunteer_invalid_lastname(self):
+        with self.assertRaises(vc.ControllerError):
+            vc.create_volunteer(username='yunsy', password='roo', firstname='Yunsy', lastname='Y',
+                                phone='+447519953189', camp=Camp(name='UCL'))
+
+    def test_create_volunteer_invalid_phone_without_international_code(self):
+        with self.assertRaises(vc.ControllerError):
+            vc.create_volunteer(username='yunsy', password='roo', firstname='Yunsy', lastname='Yin',
+                                phone='07519953189', camp=Camp(name='UCL'))
+
+    def test_create_volunteer_invalid_phone_too_short(self):
+        with self.assertRaises(vc.ControllerError):
+            vc.create_volunteer(username='yunsy', password='roo', firstname='Yunsy', lastname='Yin',
+                                phone='+44751', camp=Camp(name='UCL'))
+
+    def test_find_volunteer(self):
+        vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin', phone='+447519953189',
+                            camp=Camp(name='UCL'))
+        volunteer_called = vc.find_volunteer('yunsy')
+        self.assertEqual(type(volunteer_called), Volunteer)
+
+    def test_find_volunteer_invalid_input(self):
```

Comment:
I think naming this test `test_find_nonexistent_volunteer` would be more straightforward